### DISCUSSION TITLE
How would I pull data to visualize the x,y position of each car on track when they overtake each other?

### DISCUSSION THREAD

**Discussion by munsonbh** [theOehrly/Fast-F1#488] [Q&A] [answered]
Hello, I've been working through how to pull timing data in order to see when a car increases or decreases in grid position, and then plot when the change happens to the x,y position on track. I believe this would require `fastf1.api.timing_data` to sample grid position at closer to a 1 second interval. The closest discussion I could find was https://github.com/theOehrly/Fast-F1/discussions/198

The API is invite-only, correct? Would this be a use case that would allow me to access that data once per race?

**Comment by theOehrly** [comment, marked as answer]
Hi, you can use the ``stream_data`` that is returned by ``fastf1.api.timing_data``. That is the highest resolution of grid position data and gap to other cars that exists. 
See https://docs.fastf1.dev/api.html#fastf1.api.timing_data and also #241.

Note that the API around ``fastf1.api`` might change in the future, as noted in the documentation. It's not really intended as public API. But this specific data will then be made available through some other method. So the data itself won't go away.

Btw, I'm not sure what you mean with the API being "invite-only"? The API... well it exists. I don't own the data. From my point of view, you can more or less do whatever you want.

**Reply by munsonbh** [reply]
Thank you for your response! My confusion came from two sources:

First, when I tried to use the API, it returned an AttributeError that caused me to seek a reason:
```
AttributeError                            Traceback (most recent call last)
Cell In[27], line 3
      1 import fastf1
      2 fastf1.set_log_level('WARNING')
----> 3 fastf1.api.make_path('Abu Dhabi Grand Prix', '2023-11-26', 'Race', '2023-11-26')

AttributeError: module 'fastf1' has no attribute 'api'
```

Second, as I was trying to troubleshoot, I found this in the documentation, which I interpreted to mean that it was invite-only:
"If you currently require functionality from there, please open an issue on Github with details about what you require and why."

What I'm gathering is that the attribute error relates to my code rather than an access issue. In that case, any advice would be appreciated.

**Reply by theOehrly** [reply]
You need to explicitly import the submodule with ``import fastf1.api``.

> "If you currently require functionality from there, please open an issue on Github with details about what you require and why."

This refers to the announced deprecation and the fact that some people are using functionality from ``fastf1.api`` which will likely go away. And this was never really intended as public API. I just want to know how and why people want to interact with these more "low-level" functions directly, so I can accommodate their needs when rewriting and refactoring this. Consider this hint more as an invitation for a user survey 😅

**Reply by munsonbh** [reply]
OK, I'm getting data now, thank you!
